lib/utils.pm

```diff
@@ -258,6 +259,21 @@ sub get_netboot_mirror {
     return get_var('MIRROR_' . uc($m_protocol));
 }
 
+sub zypper_call {
+    my $command = shift;
+    my $timeout = shift || 700;
+    my $str     = bmwqemu::hashed_string("ZN$command");
+
+    script_run("zypper -n $command; echo $str-\$?- > /dev/$serialdev", 0);
+
+    my $ret = wait_serial(qr/$str-\d+-/, $timeout);
+
+    if ($ret =~ /$str-(\d+)/) {
+        return $1;
+    }
```

Comment:
This should be
```perl
if ($ret) {
  return $ret =~ /$str-(\d+)/;
}
```
wait_serial is already comparing against `qr/$str-\d+-/` so it can only fail when `$ret` is undef and original condition would then trigger undefined variable warning.
When `$ret` is set, you are guaranteed to have it in correct format.